core/shared/src/main/scala/zio/internal/ConcurrentWeakHashSet.scala

```diff
@@ -0,0 +1,579 @@
+package zio.internal
+
+import zio.internal.ConcurrentWeakHashSet.Ref
+
+import java.lang.ref.WeakReference
+import scala.collection.mutable
+import java.util.concurrent.locks.ReentrantLock
+import java.lang.ref.ReferenceQueue
+import java.util.concurrent.atomic.AtomicInteger
+
+private[zio] object ConcurrentWeakHashSet {
+
+  private[internal] val MaxConcurrencyLevel: Int     = 1 << 16
```

Comment:
So Scala will only do "constant folding" if all these `val` are marked as `final` and do not specify a type. 

So instead, write them as follows:

```suggestion
  private[internal] final val MaxConcurrencyLevel     = 1 << 16
```